File: project/.sdk/tm/rb/core/response.rb
# ProjectName SDK response

require_relative '../utility/struct/voxgig_struct'

class ProjectNameResponse
  attr_accessor :status, :status_text, :headers, :json_func, :body, :err

  def initialize(resmap = {})
    resmap ||= {}
    s = VoxgigStruct.getprop(resmap, "status")
    @status = s.is_a?(Numeric) ? s.to_i : -1
    st = VoxgigStruct.getprop(resmap, "statusText")
    @status_text = st.is_a?(String) ? st : ""
    @headers = VoxgigStruct.getprop(resmap, "headers")
    jf = VoxgigStruct.getprop(resmap, "json")
    @json_func = jf.is_a?(Proc) ? jf : nil
    @body = VoxgigStruct.getprop(resmap, "body")
    @err = VoxgigStruct.getprop(resmap, "err")
  end
end

File: project/.sdk/tm/php/core/Response.php
<?php
declare(strict_types=1);

// ProjectName SDK response

class ProjectNameResponse
{
    public int $status;
    public string $status_text;
    public mixed $headers;
    public mixed $json_func;
    public mixed $body;
    public mixed $err;

    public function __construct(array $resmap = [])
    {
        $s = \Voxgig\Struct\Struct::getprop($resmap, 'status');
        $this->status = is_numeric($s) ? (int)$s : -1;
        $st = \Voxgig\Struct\Struct::getprop($resmap, 'statusText');
        $this->status_text = is_string($st) ? $st : '';
        $h = \Voxgig\Struct\Struct::getprop($resmap, 'headers');
        $this->headers = ($h === '__UNDEFINED__') ? null : $h;
        $jf = \Voxgig\Struct\Struct::getprop($resmap, 'json');
        $this->json_func = is_callable($jf) ? $jf : null;
        $b = \Voxgig\Struct\Struct::getprop($resmap, 'body');
        $this->body = ($b === '__UNDEFINED__') ? null : $b;
        $e = \Voxgig\Struct\Struct::getprop($resmap, 'err');
        $this->err = ($e === '__UNDEFINED__') ? null : $e;
    }
}
